### ISSUE TITLE
Kayin stage Jumbo screen bug (Battle Arena Toshinden - Japan)

### ISSUE THREAD

**Issue by CometHunter92** [notaz/pcsx_rearmed#236]
Maybe I'm stupid and I'm doing something wrong, but I know that the peops plugin get the Jumbo screen working in the Battle Arena Toshinden (Japan). The standard builtin NEON plugin has the jumbo screen not working.
I read that in the last update of pcsx_rearmed the speed inconsistency got finally fixed, so I tried to compile the libretro core with the peops plugin but the core obtained has the same speed inconsistency as ever.
Is it possible to make the "peops plugin" version of the core behave as the default NEON one?
(tried the senquack gpu too and Battle Arena Toshinden plays at even faster speed with it).

**Comment by gameblabla** [comment]
A few notes 

> I read that in the last update of pcsx_rearmed the speed inconsistency got finally fixed, so I tried to compile the libretro core with the peops plugin but the core obtained has the same speed inconsistency as ever.

This is only fixed on non-ARM platforms. ARM platforms will still be affected by the bug as they use a different GTE code for it. In the ARM version, the GTE delay is halved so unfortunately this still doesn't fix the inconsistency in speed ingame.
If you want to change that, go to :
https://github.com/notaz/pcsx_rearmed/blob/master/libpcsxcore/new_dynarec/new_dynarec.c#L8820

Change
`cc+=gte_cycletab[source[i]&0x3f]/2;`
to
`cc+=gte_cycletab[source[i]&0x3f];`

My GTE fix basically only applies to x86/x86_64/PowerPC... because of that.
The C version of the GTE code can be enabled for ARM but there will be a speed penalty.
This is Battle arena toshiden without my GTE fix on x86_64 :
https://youtu.be/4wVrhOG6Mao

With my GTE fix
https://youtu.be/qUlYKjWOGog 

So basically for ARM, it will still be mostly like the former because it's getting halved.
The reason why notaz did this was because of concerns regarding potentially slowing down legitimate games.
See here (we talked about it)
https://github.com/notaz/pcsx_rearmed/pull/228

PCSX Redux fixed it differently : they handle it in their interpreter code properly. I wanted to go with that approach initially but i was able unable to get it to work properly due to complications and the switch from C++ to C.

**Comment by CometHunter92** [comment]
Explanatory. Thanks!
I'll try to compile it with the change in that line and see what happens.
See, I'm not an expert, I know that you can compile the libretro core with different gpu plugins.
The aarch64 version doesn't have the NEON gpu, but has the PEOpS one. It is slower (the gpu itself), but it is more accurate in some cases. It would be great to compile the arm32 version with the PEOpS one. I compile it with the PEOpS plugin, it works, but I don't know why, it doesn't have the full option one as the aarch64 version (maybe wrong compile settings).
For the jumbo screen in the Kayin stage, go to https://youtu.be/u0fQ56kjRK4?t=611, you can see the jumbo screen behind the characters: with the NEON gpu the screen is blank. With the PEOpS gpu the screen works.

**Comment by CometHunter92** [comment]
Now with the latest version of the core, if compiled with the peops plugin, it plays at the "right speed" and, with the peops plugin, the jumbo screen works, so it can be closed.
